packages/layout/src/footnotes/getFootNotes.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * calculates the locations  of the footnotes within the given node array
+ * this is used to check if a footnote is actually inside a node after line breaking and splitting
+ * 
+ * @param {Array} nodes 
+ * @returns Array
+ */
+function calculateFootnoteLocations(nodes) {
+  let foot_notes = [];
+  let str = '';
+  for (const item of nodes) {
+    if (item.props && item.props.footNote) {
+      foot_notes.push({ loc: str.length, el: item });
+    }
+    if (item.type == 'TEXT_INSTANCE') {
+      str += item.value;
+    }
+  }
+
+  return foot_notes;
+}
+
+/**
+ * Finds all footnotes in a given node
+ * checks if a footnote is actually present in the calculated lines of a node
+ * 
+ * @param {Object} node 
+ * @param {number} top 
+ * @returns 
+ */
+function getFootNotes(node, top = 0) {
```

Comment:
Can we refactor the places where the term `footnote` is considered as `foot note` when writing variable names?
So, here the name would be `getFootnotes` instead of `getFootNotes`